src/main/java/com/synopsys/integration/alert/workflow/filter/NotificationFilter.java

```diff
@@ -102,6 +102,34 @@ public NotificationFilter(final JsonExtractor jsonExtractor, final List<Provider
                    .collect(Collectors.toList());
     }
 
+    /**
+     * Creates a java.util.Collection of NotificationContent objects that are applicable for at least one Distribution Job.
+     * @return A java.util.List of sorted (by createdAt) NotificationContent objects.
+     */
+    public Collection<NotificationContent> extractApplicableNotifications(final Set<ProviderContentType> providerContentTypes, final CommonDistributionConfig jobConfiguration, final Collection<NotificationContent> notificationList) {
+
+        final Set<String> configuredNotificationTypes = jobConfiguration.getNotificationTypes().stream().collect(Collectors.toSet());
+        if (configuredNotificationTypes.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        final Map<String, List<NotificationContent>> notificationsByType = getNotificationsByType(configuredNotificationTypes, notificationList);
+
+        final Set<NotificationContent> filteredNotifications = new HashSet<>();
+        notificationsByType.forEach((type, groupedNotifications) -> {
+            final Set<StringHierarchicalField> filterableFields = getFilterableFieldsByNotificationType(type, providerContentTypes);
+            final Predicate<NotificationContent> filterForNotificationType = createJobFilter(filterableFields, jobConfiguration);
+
+            final List<NotificationContent> matchingNotifications = applyFilter(groupedNotifications, filterForNotificationType);
+            filteredNotifications.addAll(matchingNotifications);
+        });
+
+        return filteredNotifications
+                   .parallelStream()
```

Comment:
I don't think parallelStream() buys us anything with sorted(). I implemented it this way too, but thinking about it now doesn't make a lot of sense.